apps/io-wallet-user-func/src/infra/email.ts

```diff
@@ -0,0 +1,92 @@
+import { AuthProfileApiConfig, MailConfig } from "@/app/config";
+import { EmailNotificationService, SendEmailNotificationParams } from "@/email";
+import {
+  getMailerTransporter,
+  sendMail,
+} from "@pagopa/io-functions-commons/dist/src/mailer";
+import { EmailString, FiscalCode } from "@pagopa/ts-commons/lib/strings";
+import { flow, pipe } from "fp-ts/function";
+import * as TE from "fp-ts/lib/TaskEither";
+import * as t from "io-ts";
+
+export class EmailNotificationServiceClient
+  implements EmailNotificationService
+{
+  #authProfileApiConfig: AuthProfileApiConfig;
+  #mailConfig: MailConfig;
+
+  getUserEmail = (fiscalCode: FiscalCode) =>
+    pipe(
+      TE.tryCatch(
+        async () => {
+          const result = await fetch(
+            new URL(
+              `/api/v1/profiles/${fiscalCode}`,
+              this.#authProfileApiConfig.baseURL,
+            ),
+            {
+              headers: {
+                "x-functions-key": this.#authProfileApiConfig.apiKey,
+              },
+              method: "GET",
+              signal: AbortSignal.timeout(
+                this.#authProfileApiConfig.httpRequestTimeout,
+              ),
+            },
+          );
+          if (!result.ok) {
+            throw new Error(JSON.stringify(await result.json()));
+          }
+          return result.json();
+        },
+        (error) =>
+          new Error(
+            `Error occurred while getting user email from profile service: ${error}`,
+          ),
+      ),
+      TE.chainW(
+        flow(
+          t.type({
+            email: EmailString,
+          }).decode,
+          TE.fromEither,
+          TE.mapLeft((errors) => new Error(errors.join(", "))),
+        ),
+      ),
+      TE.map(({ email }) => email),
+    );
+
+  sendEmail = (params: SendEmailNotificationParams) =>
+    pipe(
+      // required to comply with the constraints imposed by the MailupMailerConfig type from @pagopa/io-functions-commons
+      {
+        MAIL_FROM: this.#mailConfig.mailSender,
+        MAIL_TRANSPORTS: undefined,
+        MAILHOG_HOSTNAME: undefined,
+        MAILUP_SECRET: this.#mailConfig.mailupSecret,
+        MAILUP_USERNAME: this.#mailConfig.mailupUsername,
+        NODE_ENV: "production",
+        SENDGRID_API_KEY: undefined,
+      },
+      getMailerTransporter,
+      (transporter) =>
+        sendMail(transporter, {
+          from: this.#mailConfig.mailSender,
+          html: params.html,
+          subject: params.subject,
+          text: params.text,
+          to: params.to,
```

Comment:
The `text` parameter should contain the content of the email in text format for the screen reader or for those who cannot render the HTML. It is not good to insert only the object. You can try to loose the content of the HTML as done [here](https://github.com/pagopa/io-functions-app/blob/a42f14ae2c1d944c6c8d1ea6127a2b8f395396c8/SendTemplatedLoginEmailActivity/handler.ts#L114)